packages/block-editor/src/components/inserter/media-tab/media-preview.js

```diff
@@ -167,8 +168,13 @@ export function MediaPreview( { media, onClick, category } ) {
 				.fetch( url )
 				.then( ( response ) => response.blob() )
 				.then( ( blob ) => {
+					const fileName = getFilename( url ) || 'image.jpg';
```

Comment:
Note for other reviewers: the fallback uses `.jpg` independently of the actual file format, since we rely on the server to choose an appropriate file type for us. We do need to provide *some* image extension, however, even if it's wrong, otherwise the server will outright reject the upload.